test/fail_compilation/retscope.d

```diff
@@ -235,13 +235,13 @@ void* funretscope(scope dg_t ptr) @safe
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/retscope.d(249): Error: cannot implicitly convert expression (__lambda1) of type void* delegate() pure nothrow @nogc return @safe to void* delegate() @safe
-fail_compilation/retscope.d(249): Error: cannot implicitly convert expression (__lambda1) of type void* delegate() pure nothrow @nogc return @safe to void* delegate() @safe
-fail_compilation/retscope.d(250): Error: cannot implicitly convert expression (__lambda2) of type void* delegate() pure nothrow @nogc return @safe to void* delegate() @safe
-fail_compilation/retscope.d(250): Error: cannot implicitly convert expression (__lambda2) of type void* delegate() pure nothrow @nogc return @safe to void* delegate() @safe
 ---
 */
 
+
+
+
+
```

Comment:
No you didn't. The following snippet (rightfully) doesn't compile with master, but does with your P.R.:

```D
struct Bar { int i; }

alias FunDG = Bar* delegate () @safe;

void main () @safe
{
    Bar* b = getBar();
}

Bar* getBar () @safe
{
    Bar b;
    scope FunDG x = () return { return &b; };
    return forward(x);
}

Bar* forward (scope FunDG escaper) @safe
{
    return escaper();
}
```